hack/istio/create-mesh-namespace.sh

```diff
@@ -0,0 +1,164 @@
+  -n|--namespace <name>: The namespace to be created or deleted. REQUIRED.
+  -sa|--service-accounts
+      The service accounts that will be able to authenticate with docker with the credentials you provide.
+      You can specify more than one service account separated with spaces.
+      If you are using this to be able to pull Istio addon images, this value should be "prometheus grafana default".
+      Default: "default"
+  -h|--help: this text
+HELPMSG
+      exit 1
+      ;;
+    -n|--namespace)
+      NAMESPACE="$2"
+      shift;shift
+      ;;
+    -sa|--service-accounts)
+      SERVICE_ACCOUNTS="$2"
+      shift;shift
+      ;;
+    *)
+      echo "Unknown argument [$key]. Aborting."
+      exit 1
+      ;;
+  esac
+done
+
+echo Will create namespace using these settings:
+echo CLIENT_EXE=${CLIENT_EXE}
+echo DELETE_NAMESPACE=${DELETE_NAMESPACE}
+echo ENABLE_INJECTION=${ENABLE_INJECTION}
+echo ISTIO_NAMESPACE=${ISTIO_NAMESPACE}
+echo NAMESPACE=${NAMESPACE}
+echo SERVICE_ACCOUNTS=${SERVICE_ACCOUNTS}
+
+[ -z "$NAMESPACE" ] && echo "You must specify --namespace" && exit 1
+
+IS_OPENSHIFT="false"
+IS_MAISTRA="false"
+if [[ "${CLIENT_EXE}" = *"oc" ]]; then
+  IS_OPENSHIFT="true"
+  IS_MAISTRA=$([ "$(oc get crd | grep servicemesh | wc -l)" -gt "0" ] && echo "true" || echo "false")
+fi
+
+echo "IS_OPENSHIFT=${IS_OPENSHIFT}"
+echo "IS_MAISTRA=${IS_MAISTRA}"
+
+# If we are to delete, remove everything and exit immediately after
+if [ "${DELETE_NAMESPACE}" == "true" ]; then
+  echo "Deleting namespace ${NAMESPACE}..."
+  if [ "${IS_OPENSHIFT}" == "true" ]; then
+    if [ "${IS_MAISTRA}" != "true" ]; then
+      $CLIENT_EXE delete network-attachment-definition istio-cni -n ${NAMESPACE}
+    else
+      $CLIENT_EXE delete smm default -n ${NAMESPACE}
+    fi
+    $CLIENT_EXE delete scc "${NAMESPACE}-scc"
+  fi
+  ${CLIENT_EXE} delete namespace ${NAMESPACE}
+  exit 0
+fi
+
+# Create and prepare the demo namespaces
+
+if [ "${IS_OPENSHIFT}" == "true" ]; then
+  $CLIENT_EXE new-project ${NAMESPACE}
+else
+  $CLIENT_EXE create namespace ${NAMESPACE}
+fi
+
+if [ "${ENABLE_INJECTION}" == "true" ]; then
+  ${CLIENT_EXE} label namespace ${NAMESPACE} istio-injection=enabled
+fi
+
+if [ "${IS_MAISTRA}" == "true" ]; then
+  prepare_maistra "${NAMESPACE}"
+fi
+
+if [ "${IS_OPENSHIFT}" == "true" ]; then
+  if [ "${IS_MAISTRA}" != "true" ]; then
+    cat <<NAD | $CLIENT_EXE -n ${NAMESPACE} create -f -
+apiVersion: "k8s.cni.cncf.io/v1"
+kind: NetworkAttachmentDefinition
+metadata:
+  name: istio-cni
+NAD
+  fi
+
+  USERS=""
+  for sa in ${SERVICE_ACCOUNTS}; do
+    USERS=$(printf "%s\n- system:serviceaccount:%s:%s" "${USERS}" ${NAMESPACE} ${sa})
+  done
+
+  cat <<SCC | $CLIENT_EXE apply -f -
+apiVersion: security.openshift.io/v1
+kind: SecurityContextConstraints
+metadata:
+  name: "${NAMESPACE}-scc"
+runAsUser:
+  type: RunAsAny
+  UID: <none>
+  UID Range Min: <none>
+  UID Range Max: <none>
```

Comment:
this can't possibly be right

Reply:
you took this from "oc describe" - do a "oc get" for the true yaml. But here's a hint - you don't need to specify anything regarding uid ranges.

Reply:
I just removed that, it was your dumb suggestion to begin with :)